assets/js/index.js

```diff
@@ -0,0 +1,310 @@
+      randomIndex;
+    while (currentIndex != 0) {
+      randomIndex = Math.floor(Math.random() * currentIndex);
+      currentIndex--;
+      [answersArray[currentIndex], answersArray[randomIndex]] = [
+        answersArray[randomIndex],
+        answersArray[currentIndex],
+      ];
+    }
+  }
+  return answersArray;
+};
+
+// format question text
+const formatText = (text) => {
+  if (text.includes("#039;")) {
+    text = text.replace(/&#039;/gi, "'");
+  }
+  if (text.includes("&quot;")) {
+    text = text.replace(/&quot;/gi, '"');
+  }
+  return text;
+};
+
+// start quiz
+const startQuiz = () => {
+  titleScreen.classList.add("hide");
+  questionScreen.classList.remove("hide");
+  startBtn.classList.remove("hide");
+  displayQuestions();
+};
+
+// display questions
+const displayQuestions = () => {
+  // check if there are questions left and display question with answers
+  timerItem.classList.remove("hide");
+  if (!timerStarted) {
+    timerItem.innerHTML = "Time Remaining: " + timerAmount;
+    timing = setInterval(startTimer, 1000);
+    timerStarted = true;
+  }
+  questionNumber++;
+  if (questionNumber <= questions.length && timeRemaining) {
+    getCurrentQuestion();
+    getCurrentAnswers();
+    const getAnswerContainer = document.getElementById("answer-container");
+    //getAnswerContainer.addEventListener("click", isCorrect);
+  } else {
+    gameOver();
+  }
+};
+
+// begin timer
+const startTimer = () => {
+  timer--;
+  timerItem.innerHTML = "Time Remaining: " + timer;
+  if (timer <= 0) {
+    gameOver();
+  }
+};
+
+// get current question
+const getCurrentQuestion = () => {
+  currentQuestion = questions[questionNumber - 1].question;
+
+  const questionText = document.getElementById("question-text");
+  questionText.textContent = currentQuestion;
+};
+
+// get current answers
+const getCurrentAnswers = () => {
+  // generate and create HTML elements to display answers
+  const answerContainer = document.getElementById("answer-container");
+
+  for (let i = 0; i < 4; i++) {
+    // create answer buttons
+    const answer = document.createElement("button");
+    answer.setAttribute("class", "answer btn");
+    answer.setAttribute("id", i);
+    currentAnswers = questions[questionNumber - 1].answers;
+    answer.textContent = currentAnswers[i].text;
+    answer.setAttribute("data-answer", currentAnswers[i].correct);
+    answer.addEventListener("click", isCorrect);
+    answerContainer.append(answer);
+  }
+};
+
+// check if answer is correct
+const isCorrect = (event) => {
+  const target = event.target;
+  const isAnswerCorrect = target.getAttribute("data-answer");
+
+  if (isAnswerCorrect === "true") {
+    score += scoreAmount;
+    target.classList.add("btn-correct");
+    target.classList.remove("btn");
+    setTimeout(loadNextQuestion, 500);
+  } else {
+    timer -= timePenalty;
+    target.classList.add("btn-incorrect");
+    target.classList.remove("btn");
+    correctAnswer = findCorrectAnswer();
```

Comment:
Remember to always declare variables with either `const` or `let` - in this case it is `const` since we never need to reassign the value.